sentry_sdk/client.py

```diff
@@ -334,19 +334,27 @@ def capture_event(
         if session:
             self._update_session_from_event(session, event)
 
-        if event_opt.get("type") == "transaction":
-            # Transactions should go to the /envelope/ endpoint.
-            self.transport.capture_envelope(
-                Envelope(
-                    headers={
-                        "event_id": event_opt["event_id"],
-                        "sent_at": format_timestamp(datetime.utcnow()),
-                    },
-                    items=[
-                        Item(payload=PayloadRef(json=event_opt), type="transaction"),
-                    ],
-                )
+        attachments = hint.get("attachments")
+        is_transaction = event_opt.get("type") == "transaction"
+
+        if is_transaction or attachments:
+            # Transactions or events with attachments should go to the
+            # /envelope/ endpoint.
+            envelope = Envelope(
+                headers={
+                    "event_id": event_opt["event_id"],
+                    "sent_at": format_timestamp(datetime.utcnow()),
+                }
             )
+
+            if is_transaction:
+                envelope.add_transaction(event_opt)
+            else:
+                envelope.add_event(event_opt)
```

Comment:
@rhcarvalho I'm starting to think this entire forcing events and transactions to go different paths on the envelope to be wrong-ish. We now need to check if it's a transaction or not to call a different method on the envelope but in both cases they do the same internally. The only difference is that `add_event` and `add_transaction` pass different event types.